File: code_challenge_solution/java/157 ROT 13.java
```java
public class ROT13Cipher {

    public static void main(String[] args) {
        String message = "Hello World";
        String encryptedMessage = encrypt(message);
        System.out.println("Encrypted Message: " + encryptedMessage);
        
        String decryptedMessage = decrypt(encryptedMessage);
        System.out.println("Decrypted Message: " + decryptedMessage);
    }

    public static String encrypt(String message) {
        StringBuilder encryptedMessage = new StringBuilder();
        for (char c : message.toCharArray()) {
            if (Character.isUpperCase(c)) {
                encryptedMessage.append((char) ((c - 'A' + 13) % 26 + 'A'));
            } else if (Character.isLowerCase(c)) {
                encryptedMessage.append((char) ((c - 'a' + 13) % 26 + 'a'));
            } else {
                encryptedMessage.append(c);
            }
        }
        return encryptedMessage.toString();
    }

    public static String decrypt(String encryptedMessage) {
        return encrypt(encryptedMessage); // ROT13 is self-inverse, so same function can be used for decryption
    }
}
```

File: code_challenge_solution/c#/157 ROT 13.cs
using System;

class Program
{
    static void Main(string[] args)
    {
        string input = "Hello, World!";
        string output = ROT13(input);
        Console.WriteLine(output);
    }

    static string ROT13(string input)
    {
        char[] array = input.ToCharArray();

        for (int i = 0; i < array.Length; i++)
        {
            if (char.IsLetter(array[i]))
            {
                char baseChar = char.IsUpper(array[i]) ? 'A' : 'a';
                array[i] = (char)((array[i] - baseChar + 13) % 26 + baseChar);
            }
        }

        return new string(array);
    }
}